Mods/src/PuppiMod.cc

```diff
@@ -141,235 +160,222 @@ void PuppiMod::SlaveBegin()
     }
   }
   fNumEtaBins = fMaxEtas.size();
-  std::cout << "Number of Eta bins: " << fNumEtaBins << std::endl;
+  Info("SlaveBegin", "Number of Eta bins: %d", fNumEtaBins);
+
+  fMapper = new ParticleMapper(fR0, fR0, fMaxEtas[fNumEtaBins-1]);
 }
 
 //--------------------------------------------------------------------------------------------------
-void PuppiMod::SlaveTerminate()
+void
+PuppiMod::SlaveTerminate()
 {
   // ===== deallocate memory ====
-  // Or you know, don't
+  delete fPuppiParticles;
+  delete fMapper;
 }
 
 //--------------------------------------------------------------------------------------------------
-void PuppiMod::Process()
+void
+PuppiMod::Process()
 {
-  // Process entries of the tree. 
-  fVertexes = GetObject<VertexCol>(fVertexesName);
-  fPFCandidates = GetObject<PFCandidateCol>(fPFCandidatesName);
+  // Process entries of the tree.
+  auto* vertexes = GetObject<VertexCol>(fVertexesName);
+  if (!vertexes)
+    SendError(kAbortAnalysis, "Process", "PV collection not found");
```

Comment:
Make sure you didn't get a null pointer! Seg fault is much more annoying than job abort.